File: app/src/main/java/br/com/alura/ceep/ui/coffemachine/presentation/view/custom/CoffeAdapter.java
package br.com.alura.ceep.ui.coffemachine.presentation.view.custom;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import br.com.alura.ceep.ui.coffemachine.R;
import br.com.alura.ceep.ui.coffemachine.domain.CoffeMachineData;
import br.com.alura.ceep.ui.coffemachine.presentation.view.DetailActivity;
import org.jetbrains.annotations.NotNull;

public class CoffeAdapter extends RecyclerView.Adapter<CoffeeViewHolder> {

  private Context context;
  CoffeMachineData[] coffeMachineData;

  public CoffeAdapter(CoffeMachineData[] coffeMachineData) {
    this.coffeMachineData = coffeMachineData;
  }

  @NonNull
  @NotNull
  @Override
  public CoffeeViewHolder onCreateViewHolder(@NonNull @NotNull ViewGroup parent, int viewType) {
    context = parent.getContext();
    LayoutInflater layoutInflater = LayoutInflater.from(context);
    View view = layoutInflater.inflate(R.layout.coffe_item_list, parent, false);
    return new CoffeeViewHolder(view);
  }

  public void onBindViewHolder(@NonNull @NotNull CoffeeViewHolder holder, int position) {
    holder.itemView.setOnClickListener(v -> {
      Bundle bundle = new Bundle();
      bundle.putSerializable("coffe", coffeMachineData[position]);
      bundle.putString("nome", coffeMachineData[position].getCoffeDescription());
      Intent intent = new Intent(context, DetailActivity.class);
      intent.putExtras(bundle);
      context.startActivity(intent);
    });
    final CoffeMachineData coffeMachineDataList = coffeMachineData[position];
    holder.textViewDescription.setText(coffeMachineDataList.getCoffeDescription());
    holder.textViewSize.setText(coffeMachineDataList.getCoffeSize());
    holder.textQtd.setText(coffeMachineDataList.getCoffeSizeTitule());
    holder.textViewIntensity.setText(coffeMachineDataList.getCoffeIntensity());
    holder.textInt.setText(coffeMachineDataList.getCoffeIntensityTitule());
    holder.textViewType.setText(coffeMachineDataList.getCoffeType());
    holder.coffeImage.setImageResource(coffeMachineDataList.getCoffeImage());
  }

  @Override
  public int getItemCount() {
    return coffeMachineData.length;
  }
}

File: app/src/main/java/br/com/alura/ceep/ui/coffemachine/presentation/view/custom/CoffeAdapter.kt
package br.com.alura.ceep.ui.coffemachine.presentation.view.custom

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.content.ContextCompat.startActivity
import androidx.recyclerview.widget.RecyclerView
import br.com.alura.ceep.ui.coffemachine.R
import br.com.alura.ceep.ui.coffemachine.domain.CoffeMachineData
import br.com.alura.ceep.ui.coffemachine.presentation.view.DetailActivity

class CoffeAdapter(var coffeMachineData: Array<CoffeMachineData>) :
    RecyclerView.Adapter<CoffeeViewHolder>() {
    private var context: Context? = null
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CoffeeViewHolder {
        context = parent.context
        val layoutInflater = LayoutInflater.from(context)
        val view = layoutInflater.inflate(R.layout.coffe_item_list, parent, false)
        return CoffeeViewHolder(view)
    }

    override fun onBindViewHolder(holder: CoffeeViewHolder, position: Int) {
        holder.itemView.setOnClickListener { v: View? ->
            val bundle = Bundle()
            bundle.putSerializable("coffe", coffeMachineData[position])
            bundle.putString("nome", coffeMachineData[position].CoffeDescription)
            val intent = Intent(context, DetailActivity::class.java)
            intent.putExtras(bundle)
            context?.startActivity(intent)
        }
        val coffeMachineDataList = coffeMachineData[position]
        holder.textViewDescription.text = coffeMachineDataList.CoffeDescription
        holder.textViewSize.text = coffeMachineDataList.CoffeSize
        holder.textQtd.text = coffeMachineDataList.CoffeSizeTitule
        holder.textViewIntensity.text = coffeMachineDataList.CoffeIntensity
        holder.textInt.text = coffeMachineDataList.CoffeIntensityTitule
        holder.textViewType.text = coffeMachineDataList.Coffetype
        holder.coffeImage.setImageResource(coffeMachineDataList.CoffeImage)
    }

    override fun getItemCount(): Int {
        return coffeMachineData.size
    }
}